src/v0/destinations/clevertap/transform.js

```diff
@@ -79,12 +79,21 @@ const responseWrapper = (payload, destination) => {
  * @returns
  * return the final payload after converting to the relevant data-types.
  */
-const convertObjectAndArrayToString = (payload) => {
+const convertObjectAndArrayToString = (payload, event) => {
   const finalPayload = {};
   if (payload) {
     Object.keys(payload).forEach((key) => {
       if (payload[key] && (Array.isArray(payload[key]) || typeof payload[key] === 'object')) {
-        finalPayload[key] = JSON.stringify(payload[key]);
+        if (
+          event === 'Charged' &&
+          key === 'Items' &&
+          Array.isArray(payload[key]) &&
+          typeof payload[key][0] === 'object'
+        ) {
+          finalPayload[key] = payload[key];
+        } else {
+          finalPayload[key] = JSON.stringify(payload[key]);
+        }
```

Comment:
Rather than checking for each element, we can check outside of the for loop and if the event is charged and the key is items we can non-stringify that.

Reply:
But in this way we can't check if the Items is an array of objects or not.

Reply:
After non-stringifying we can check right?

Reply:
Done